src/components/logic/always_after.py

```diff
@@ -0,0 +1,40 @@
+""" Algorithm to detect the always after relationships in a given log """
+import os
+from src.components.logic.relationship import Relationship
+from src.components.util.xes_importer import XES_Importer
+from pm4py.algo.discovery.log_skeleton import algorithm as lsk_discovery
+
+
+class AlwaysAfter(Relationship):
+    """ Implementation of the always after relationship"""
+
+    def __init__(self, traces):
+        super().__init__(traces)
+
+    def activity_pair_matches(self, trace, activity1, activity2) -> bool:
+        """Determine if the given pair of activities in the always after condition."""
+
+        # Non-reflexive
+        if activity1 == activity2:
+            return False
+
+        activity_projection = self.project_trace(trace, [activity1, activity2])
+
+        if activity1 not in activity_projection:
+            return True
+        elif activity_projection[-1] == activity2:
+            return True
+        else:
+            return False
```

Comment:
Can you change that bit of code to:

`return activity1 not in activity_projection or activity_projection[-1] == activity2`

I think this is much cleaner. 😄